src/plugins/intel_gpu/src/plugin/transformations/fc_horizontal_fusion.cpp

```diff
@@ -299,6 +299,67 @@ FullyConnectedHorizontalFusion::FullyConnectedHorizontalFusion(bool fuse_mlp_swi
             }
             org_fc->clear_control_dependencies();
         }
+
+        // Merge scalar multiply layers into one when all scalar constants have the same value.
+        //
+        //          FusedFC                     FusedFC
+        //             |                           |
+        //       VariadicSplit      ==>         new_Mul  (to be fused with FusedFC)
+        //      /      |      \                    |
+        //    Mul     Mul     Mul            VariadicSplit
+        //     |       |       |             |     |     |
+        const auto is_scalar_const = [](const ov::Output<ov::Node>& output) -> bool {
+            if (!ov::is_type<ov::op::v0::Constant>(output.get_node()))
+                return false;
+            const auto shape = output.get_partial_shape();
+            if (shape.is_dynamic())
+                return false;
+            return ov::shape_size(shape.to_shape()) == 1;
+        };
+
+        std::vector<float> const_values;
+        bool can_be_merged = true;
+        std::shared_ptr<ov::op::v0::Constant> const_node = nullptr;
+        for (auto& output : output_split->outputs()) {
+            auto target_node = output.get_target_inputs().begin()->get_node();
```

Comment:
I think this is not safe in case when out port has no users. So I'd recommend to check users count first (that it's equal to 1), and then check the consumer op type

Reply:
I updated it as you reviewed. Thank you!